storage/series/series.go

```diff
@@ -597,10 +597,45 @@ func (s *dbSeries) Flush(
 	if err != nil {
 		return err
 	}
-
 	return persistFn(s.id, segment, b.Checksum())
 }
 
+func (s *dbSeries) Snapshot(
+	ctx context.Context,
+	blockStart time.Time,
+	persistFn persist.Fn,
+) error {
+	s.RLock()
+	defer s.RUnlock()
+
+	if s.bs != bootstrapped {
+		return errSeriesNotBootstrapped
+	}
+
+	var (
+		buckets  = s.buffer.Snapshot(ctx, blockStart)
```

Comment:
would `readers` be a better name than buckets here?

Reply:
yeah

Reply:
done